core/src/main/java/org/apache/calcite/sql/SqlCollectionTypeNameSpec.java

```diff
@@ -58,32 +58,51 @@
  */
 public class SqlCollectionTypeNameSpec extends SqlTypeNameSpec {
   private final SqlTypeNameSpec elementTypeName;
+  private final boolean elementTypeNullable;
   private final SqlTypeName collectionTypeName;
 
   /**
    * Creates a {@code SqlCollectionTypeNameSpec}.
    *
    * @param elementTypeName    Type of the collection element
+   * @param elementTypeNullable Type of the collection element is nullable
    * @param collectionTypeName Collection type name
    * @param pos                Parser position, must not be null
    */
   public SqlCollectionTypeNameSpec(SqlTypeNameSpec elementTypeName,
+      boolean elementTypeNullable,
       SqlTypeName collectionTypeName,
       SqlParserPos pos) {
     super(new SqlIdentifier(collectionTypeName.name(), pos), pos);
     this.elementTypeName = requireNonNull(elementTypeName, "elementTypeName");
+    this.elementTypeNullable = elementTypeNullable;
     this.collectionTypeName =
         requireNonNull(collectionTypeName, "collectionTypeName");
   }
 
+  /**
+   * Creates a {@code SqlCollectionTypeNameSpec}.
+   *
+   * @param elementTypeName    Type of the collection element
+   * @param collectionTypeName Collection type name
+   * @param pos                Parser position, must not be null
+   */
+  public SqlCollectionTypeNameSpec(SqlTypeNameSpec elementTypeName,
+      SqlTypeName collectionTypeName,
+      SqlParserPos pos) {
+    this(elementTypeName, true, collectionTypeName, pos);
+  }
+
   public SqlTypeNameSpec getElementTypeName() {
     return elementTypeName;
   }
 
   @Override public RelDataType deriveType(SqlValidator validator) {
     RelDataType type = elementTypeName.deriveType(validator);
     // We have to assume that elements may be nullable
```

Comment:
The code comments need to be updated.

Reply:
What's wrong with this comment? For me it still looks valid.

Reply:
In my opinion, the current comments indicate that they are always nullable, but the current implementation is based on the "elementTypeNullable" property, which is different.This depends on whether you want to make the change or not.

Reply:
Agree. Ideally, you shouldn’t have two constructors with the same initial sentence.